### DISCUSSION TITLE
How can I use the X Window System X11 on Watcom?

### DISCUSSION THREAD

**Discussion by phikill** [open-watcom/open-watcom-v2#1181] [Q&A]
I want to develop applications with x11, but the x11 static library was compiled by gcc and watcom cannot read these objects.

and I have no idea how to compile X11 using Open Watcom.

**Comment by winspool** [comment]
For your X11 client application,
do you use "xlib.h" or the lower level "xcb.h"?

**Comment by jmalak** [comment]
You need to port X11 to OW because gcc compiled static library version is not compiled against OW C run-time library but probably against GNU C library.
I also think that X11 is not fully C standard compliant and therefore original X11 code will need to be modified to be compilable with OW.
Take into acount that OW C run-time library doesn't support full POSIX standard only basic functions that some POSIX functions used by X11 could not be available in OW (must be first implemented in OW CRTL).
